extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/DevConsoleProcessor.java

```diff
@@ -432,7 +433,7 @@ private void collectTemplates(BuildProducer<DevTemplatePathBuildItem> devTemplat
                         jarPath = jarPath.substring(1).replace('/', '\\');
                     }
                     try (FileSystem fs = FileSystems
-                            .newFileSystem(Paths.get(jarPath), classLoader)) {
+                            .newFileSystem(Paths.get(URLDecoder.decode(jarPath, StandardCharsets.UTF_8.name())), classLoader)) {
```

Comment:
Not completely sure about this TBH. It works properly but I wonder if instead we should be reconstructing a URI